### DISCUSSION TITLE
Understanding the coordinate space for Lines vs Polygons

### DISCUSSION THREAD

**Discussion by davobutt** [excaliburjs/Excalibur#2729] [Q&A] [answered]
Just beginning using excalibur and I have a question about drawing..
In my actor I'm using polygon to draw a hexagon:

```javascript
this.graphics.use(
      new Polygon({
        points: [vec(0, 50), vec(-43, 25), vec(-43, -25), vec(0, -50), vec(43, -25), vec(43, 25), vec(0, 50)],
        color: Color.Green,
      })
    );
```
Say I want to outline the polygon using the Line graphic.. it doesn't seem to obey the same coordinate space...

```javascript
const line = new Line({ start: vec(0, 50), end: vec(-43, 25), color: Color.Blue });
```

When rendered it looks like this

![image](https://github.com/excaliburjs/Excalibur/assets/29703925/9a17a3a7-b0b4-48b6-86fe-4d885d4f4fe4)

It seems to centre the line on the end coordinate?
Obviously I'm missing something, can someone point me in the right direction?

**Comment by eonarheim** [comment, marked as answer]
Hi @davobutt this is a great question! 

Graphics do indeed live in their own local space and are completely separate from each other.

The actor graphics component centers by default around around the actor's position. The property that controls this is the `ex.Actor.graphics.anchor` which is set to `(0.5, 0.5)` by default. The tricky thing about layering graphics like `ex.Line` and `ex.Polygon` is it is currently done using their bounds.

If you align the graphics to the top left corner you can layer them (but requires some tweaks to the original math)

```typescript
hexagon.graphics.anchor = ex.vec(0, 0);
hexagon.graphics.use(
  new ex.Polygon({
    points: [
      ex.vec(0, 50),
      ex.vec(-43, 25),
      ex.vec(-43, -25),
      ex.vec(0, -50),
      ex.vec(43, -25),
      ex.vec(43, 25),
      ex.vec(0, 50)],
    color: ex.Color.Green,
  })
);
var line = new ex.Line({
  start: ex.vec(43, 100),
  end: ex.vec(0, 75),
  color: ex.Color.Blue,
  thickness: 2 });
hexagon.graphics.show(line);
```

<img width="76" alt="image" src="https://github.com/excaliburjs/Excalibur/assets/612071/888764b6-20d5-48de-a0a1-cf4dbd51cdbf">


If your goal is to get and outline on the hexagon you can also do this with the `ex.Polygon`:

```typescript
hexagon.graphics.use(
  new ex.Polygon({
    strokeColor: ex.Color.Blue,
    lineWidth: 2,
    padding: 2,
    points: [
      ex.vec(0, 50),
      ex.vec(-43, 25),
      ex.vec(-43, -25),
      ex.vec(0, -50),
      ex.vec(43, -25),
      ex.vec(43, 25),
      ex.vec(0, 50)],
    color: ex.Color.Green,
  })
);
```
<img width="68" alt="image" src="https://github.com/excaliburjs/Excalibur/assets/612071/ec83e26c-8bc8-437c-9e7b-705352342e58">

You can also use the `ex.Canvas` graphic to do more complicated drawing on a Canvas2d context if need be

```typescript
const canvas = new ex.Canvas({
    width: 200,
    height: 200,
    cache: true,  // If true draw once until flagged dirty again, otherwise draw to Canvas every frame
    draw: (ctx) => {
        console.log('With cache=true I draw once');
        ctx.fillStyle = 'red';
        ctx.fillRect(0, 0, 200, 200);
    }
})

actor.graphics.use(canvas);
```
<img width="171" alt="image" src="https://github.com/excaliburjs/Excalibur/assets/612071/c869b936-8d11-4228-9f9c-42bf2be2b423">

**Reply by davobutt** [reply]
Thanks for taking the time to reply! Great answer
